model_analyzer/reports/report_manager.py

```diff
@@ -0,0 +1,223 @@
+                f" a total available memory of {max_memories} respectively.")
+
+            summary.add_paragraph(
+                "The following table summarizes each configuration at the measurement"
+                " that optimizes the desired metrics under the given constraints in"
+                " decreasing order of throughput.")
+            summary.add_table(table=self._build_summary_table(
+                model_name=model_name))
+        return summary
+
+    def _build_summary_table(self, model_name):
+        """
+        Creates a result table corresponding
+        to the best measurements for a particular
+        model
+        """
+
+        summary_table = ResultTable(headers=[
+            'Model Config Name', 'Dynamic Batcher Sizes', 'Instance Count',
+            'p99 Latency (ms)', 'Throughput (infer/sec)',
+            'GPU Memory Usage (MB)', 'GPU Utilization (%)'
+        ],
+                                    title="Report Table")
+
+        for model_config, measurement in self._measurements[model_name]:
+            if 'dynamic_batching' in model_config.get_config():
+                dynamic_batch_sizes = model_config.get_config(
+                )['dynamic_batching']['preferred_batch_size']
+                dynamic_batching_str = ','.join(
+                    [str(x) for x in dynamic_batch_sizes])
+            else:
+                dynamic_batching_str = "Disabled"
+            instance_group_list = model_config.get_config()['instance_group']
+            group_str_list = [
+                f"{group['count']}-{group['kind'].split('_')[1]}"
+                for group in instance_group_list
+            ]
+            instance_group_str = ','.join(group_str_list)
+            row = [
+                model_config.get_field('name'), dynamic_batching_str,
+                instance_group_str,
+                measurement.get_value_of_metric('perf_latency').value(),
+                measurement.get_value_of_metric('perf_throughput').value(),
+                measurement.get_value_of_metric('gpu_used_memory').value(),
+                measurement.get_value_of_metric('gpu_utilization').value()
+            ]
+            summary_table.insert_row_by_index(row)
+        return summary_table
+
+    def _get_gpu_stats(self, model_name):
+        """
+        Gets names and memory infos 
+        of GPUs used in best measurements
+        """
+
+        gpu_names = []
+        max_memories = []
+        seen_gpus = set()
+
+        for _, measurement in self._measurements[model_name]:
+            for gpu in cuda.gpus:
+                if gpu.id in measurement.gpus_used(
+                ) and gpu.id not in seen_gpus:
+                    seen_gpus.add(gpu.id)
+                    gpu_names.append((gpu.name).decode('ascii'))
+                    with gpu:
+                        mems = cuda.current_context().get_memory_info()
+                        max_memories.append(mems.total // (2**30))
```

Comment:
What does 2^30 represent? It looks like a very large number

Reply:
I'll add a comment. Its to convert bytes to GB